tensordict/nn/common.py

```diff
@@ -38,6 +38,23 @@ class FunctionalModuleWithBuffers:
 ]
 
 
+def _is_iterable_of_str(sequence_of_str: Sequence[str]) -> bool:
```

Comment:
is not equivalent

Reply:
Why? What could you do to make one return something different than the other?
My point is that if / else is slower than try/except if the except is not encountered often, which is the case here I think. Plus we can avoid some extra checks.

Reply:
basically everything that is "empty": (), tuple(""), [("", "")]

Reply:
Fine for the empty tuples, but do we want to enforce non-empty strings?
It's not ideal but at that point I think it's the user responsibility to choose key names that are appropriate. 
If the error is "you don't have a list of tuple of strings" but actually they do it's going to be weird no?
I'm not much in favour of preventing empty strings in tensordict, I think it's an unnecessary check and string comparison should be kept at its minimum to avoid the overhead.

Reply:
I'm not sure I would advice people to do it but I would not prevent it without a clear reason to do it.
My point is regarding the string comparison overhead rather than the appropriateness of this usage.
I think that this 
```python
def _is_iterable_of_str(sequence_of_str: Sequence[str]) -> bool:
    try:
        return len(sequence_of_str) and all(isinstance(key, str) or _is_iterable_of_str(key) for key in sequence_of_str)
    except Exception:
        return False

```

is still the fastest option